cvmfs/file_watcher_kqueue.cc

```diff
@@ -0,0 +1,164 @@
+/**
+ * This file is part of the CernVM File System.
+ */
+
+#include <fcntl.h>
+#include <sys/event.h>
+#include <sys/time.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+#include <string>
+#include <vector>
+
+#include "backoff.h"
+#include "file_watcher_kqueue.h"
+#include "logging.h"
+#include "util/posix.h"
+
+namespace file_watcher {
+
+FileWatcherKqueue::FileWatcherKqueue() {}
+
+FileWatcherKqueue::~FileWatcherKqueue() {}
+
+bool FileWatcherKqueue::RunEventLoop(const FileWatcher::HandlerMap& handlers,
+                                     int control_pipe) {
+  kq_ = kqueue();
```

Comment:
I think the file descriptor is never closed.

Reply:
They are now at file_watcher_kqueue.cc:115